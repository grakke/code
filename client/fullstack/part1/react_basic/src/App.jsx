import { useState, React } from 'react'

import './App.css'
import Course from './components/Course'

const Hello = ({ name, age }) => {
  const bornYear = () => new Date().getFullYear() - age

  return (
    <div>
      <p>
        Hello {name}, you are {age} years old
      </p>
      <p>So you were probably born in {bornYear()}</p>
    </div>
  )
}

const Display = ({ count }) => <div>count is {count}</div>
const Button = ({ onClick, text }) =>
  <button onClick={onClick}>
    {text}
  </button>

const History = (props) => {
  if (props.allClicks.length === 0) {
    return (
      <div>
        the app is used by pressing the buttons
      </div>
    )
  }
  return (
    <div>
      Button press history: {props.allClicks.join(' ')}
    </div>
  )
}

const Footer = () => {
  return (
    <div>
      greeting app created by <a href='https://github.com/grakke'>henry</a>
    </div>
  )
}

const App = () => {
  const now = new Date()
  const a = 10
  const b = 20
  console.log(now, a + b)

  const name = 'Peter'
  const age = 37
  const friends = ['Peter', 'Maya']

  const courses = [{
    id: 1,
    name: 'Half Stack application development',
    parts: [
      {
        name: 'Fundamentals of React',
        exercises: 10,
        id: 1
      },
      {
        name: 'Using props to pass data',
        exercises: 7,
        id: 2
      },
      {
        name: 'State of a component',
        exercises: 14,
        id: 3
      },
      {
        name: 'Redux',
        exercises: 11,
        id: 4
      }
    ]
  },
    {
      name: 'Node.js',
      id: 2,
      parts: [
        {
          name: 'Routing',
          exercises: 3,
          id: 1
        },
        {
          name: 'Middlewares',
          exercises: 7,
          id: 2
      }
    ]
  }
  ]

  const [count, setCount] = useState(0)

  const increaseByOne = () => setCount(count + 1)
  const decreaseByOne = () => setCount(count - 1)
  const setToZero = () => setCount(0)

  const [clicks, setClicks] = useState({
    left: 0, right: 0
  })
  const [allClicks, setAll] = useState([])
  const [total, setTotal] = useState(0)

  const handleLeftClick = () => {
    setAll(allClicks.concat('L'))
    const updatedLeft = clicks.left + 1
    setClicks({ ...clicks, left: updatedLeft })
    setTotal(updatedLeft + clicks.right)
  }
  const handleRightClick = () => {
    setAll(allClicks.concat('R'))
    const updateRight = clicks.right + 1
    setClicks({ ...clicks, right: updateRight })
    setTotal(clicks.left + updateRight)
  }

  return (
    <>
      <div>
        <h2>Direct Rendering:Greetings </h2>
        <p>Hello world, it is {now.toString()}</p>
        <p>
          {a} plus {b} is {a + b}
        </p>
        <Hello name={name} age={age} />
        <p>{friends}</p>
      </div>

      {courses.map(course =>
        <div id={course.id}><Course course={course} /></div>)}

      <Footer />
      <div className="card">
        <h2>stateful</h2>
        <Display count={count} />
        <Button onClick={increaseByOne} text='plus' />
        <Button onClick={setToZero} text='zero' />
        <Button onClick={decreaseByOne} text='minus' />
      </div>

      <div>
        <h2>A more complex state</h2>
        {clicks.left}
        <Button onClick={handleLeftClick} text='left' />
        <Button onClick={handleRightClick} text='right' />
        {clicks.right}
        <p>{allClicks.join(' ')}</p>
        <p>Total: {total}</p>
        <History allClicks={allClicks} />
      </div>
    </>
  )
}

export default App
