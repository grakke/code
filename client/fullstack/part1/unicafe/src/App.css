.App {
    text-align: center;
}

button {
    margin: 5px;
}

#anecdote {
    height: 60px;
}
