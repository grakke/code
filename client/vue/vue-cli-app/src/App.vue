<template>
  <div id="app">
    <img alt="Vue logo"
         src="./assets/logo.png" />

    <HelloWorld msg="Welcome to Your Vue.js App" />

    <message>Test Apple</message>
    <message>Test Oooo</message>
    <hr />

    <!-- Button trigger modal -->
    <button type="button"
            class="btn btn-primary"
            data-toggle="modal"
            data-target="#exampleModal">模态框</button>

    <!-- Modal -->
    <ModalExample :languages="this.languages">
      <template slot="header">{{ this.title }}</template>
      <template scope="bodyProps">
        {{ bodyProps.language.name }}
        <span v-if="bodyProps.language.name === 'PHP'">☑️</span>
      </template>
    </ModalExample>

    <div>
      <router-link to="/page1">Go to Page 1</router-link>
      <router-link to="/page2">Go to Page 2</router-link>
    </div>
    <router-view />

  </div>
</template >

<script>
import HelloWorld from './components/HelloWorld.vue'
import Message from './components/Message.vue'
import ModalExample from './components/ModalExample.vue'

export default {
  name: 'App',
  data() {
    return {
      title: 'Web 编程语言',
      languages: [
        { id: 1, name: 'PHP' },
        { id: 2, name: 'JavaScript' },
        { id: 3, name: 'Golang' },
      ],
    }
  },

  // 局部注册（注册到指定父组件）
  components: {
    HelloWorld,
    Message,
    ModalExample,
  },
}
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>

<style lang="scss">
@import './styles/_variables.scss';
@import './styles/_reset.scss';
@import './styles/_global.scss';
</style>