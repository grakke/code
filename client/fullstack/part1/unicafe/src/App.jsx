import { useState, React } from 'react'

import './App.css'

const Button = (props) => (
  <button onClick={props.handleClick}>
    {props.text}
  </button>
)
const StatisticLine = ({ name, score }) => (
  <tr>
    <td width='140px'>
      {name}
    </td>
    <td>
      {score}
    </td>
  </tr>
)

const Statistics = ({ good, bad, neutral }) => {
  let all = good + bad + neutral
  let average = (good + bad * -1) / all
  average = !isNaN(average) ? average : 0
  let positive = good ? (good / all) * 100 + '%' : 0

  if (all === 0) {
    return `No feedback given`
  }

  return (
    <div>
      <h2>statistics</h2>
      <table>
        <StatisticLine name='good' score={good} />
        <StatisticLine name='neutral' score={neutral} />
        <StatisticLine name='bad' score={bad} />
        <StatisticLine name='all' score={all} />
        <StatisticLine name='average' score={average} />
        <StatisticLine name='positive' score={positive} />
      </table>
    </div>
  )
}
const points = [0, 0, 7, 11, 8, 9, 0, 99]

const App = () => {
  const [good, setGood] = useState(0)
  const [neutral, setNeutral] = useState(0)
  const [bad, setBad] = useState(0)

  const anecdotes = [
    'If it hurts, do it more often.',
    'Adding manpower to a late software project makes it later!',
    'The first 90 percent of the code accounts for the first 90 percent of the development time...The remaining 10 percent of the code accounts for the other 90 percent of the development time.',
    'Any fool can write code that a computer can understand. Good programmers write code that humans can understand.',
    'Premature optimization is the root of all evil.',
    'Debugging is twice as hard as writing the code in the first place. Therefore, if you write the code as cleverly as possible, you are, by definition, not smart enough to debug it.',
    'Programming without an extremely heavy use of console.log is same as if a doctor would refuse to use x-rays or blood tests when diagnosing patients.',
    'The only way to go fast, is to go well.'
  ]
  const [selected, setSelected] = useState(0)
  const nextIndex = parseInt(Math.random() * anecdotes.length, 10)
  console.log(nextIndex)
  const handleVote = () => {
    // const copy = { ...points }
    // copy[2] += 1
  }
  const mostVotesCount = () => {
    let index = 0
    for (let i = 1; i < anecdotes.length; i++) {
      if (points[i] >= points[index]) {
        index = i
      }
    }
    return index
  }
  const mostVotesIndex = mostVotesCount()
  console.log(mostVotesIndex)
  return (
    <div>
      <div>
        <h2>give feedback</h2>
        <div>
          <Button handleClick={() => setGood(good + 1)} text="good" />
          <Button handleClick={() => setNeutral(neutral + 1)} text="neutral" />
          <Button handleClick={() => setBad(bad + 1)} text="bad" />
        </div>

        <Statistics good={good} bad={bad} neutral={neutral} />
      </div>

      <div>
        <h2>Anecdote of the Day</h2>
        <div id='anecdote'>
          {anecdotes[selected]}
        </div>
        <div id='anecdote'>
          <p> has {points[selected]} votes</p>
        </div>
        <Button handleClick={handleVote} text="Vote" index={selected} />
        <Button handleClick={() => setSelected(nextIndex)} text="Next anecdote" />

        <h2>Anecdote of the most votes</h2>
        <div id='anecdote'>
          {anecdotes[mostVotesIndex]}
        </div>
        <div id='anecdote'>
          <p> has {points[mostVotesIndex]} votes</p>
        </div>
      </div>
    </div>
  )
}

export default App