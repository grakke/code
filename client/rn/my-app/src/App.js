import { useState } from "react";
import './App.css';

function App(props) {
    // state = {
    //     isActive: false
    // }
    const [status, setStatus] = useState(false);

    function handleClick() {
        setStatus(!status)
    }

    return (

        <div className="App">
            <button onClick={handleClick}>
                {status ? "Hide" : "Show"}
            </button>
            {/* <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <p>
                    Edit <code>src/App.js</code> and save to reload.
                </p>
                <a
                    className="App-link"
                    href="https://reactjs.org"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    Learn React
                </a>
            </header> */}
        </div >
    );
}

export default App;
