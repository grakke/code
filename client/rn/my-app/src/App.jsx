import { BrowserRouter, Route, Switch } from 'react-router-dom';
import './App.css';
import LandingPage from './pages/landingPage';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Switch>
          <Route path="/" component={LandingPage} />
        </Switch>
      </BrowserRouter>
    </div>
  );
}

export default App;