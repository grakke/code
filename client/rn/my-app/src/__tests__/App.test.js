import React from 'react';
import { create } from 'react-test-renderer';
import App from '../App';

describe('My first snapshot test', () => {
    test('testing app button', () => {
        let tree = create(<App />)
        expect(tree.toJSON()).toMatchSnapshot();
    })
})

describe("Changing our button name to Hide", () => {

    test('toggle the button', () => {
        let tree = create(<App />);

        let instance = tree.getInstance();

        // TODO:Changing our button name to Hide â€º toggle the button TypeError: Cannot read property 'state' of null
        expect(instance.state.isActive).toBe(false)

        // changing  the state
        instance.handleClick();

        // isActive property is updated to `true`
        expect(instance.state.isActive).toBe(true);

        expect(tree.toJSON()).toMatchSnapshot()
    })
})
