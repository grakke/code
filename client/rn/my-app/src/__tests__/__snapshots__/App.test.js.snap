// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing app button 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    Show
  </button>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
